library/include/modules/Materials.h

```diff
@@ -78,6 +78,7 @@ namespace DFHack
 
         int16_t type;
         int32_t index;
+        df::item_type itype;
```

Comment:
I'm a bit reluctant to introduce item-specific state in a material-specific class (I think this is basically what you brought up in https://github.com/DFHack/dfhack/issues/1699#issuecomment-726472912). The only place I see this state used is in Job::isSuitableMaterial(), specifically when the MaterialInfo is passed to `iinfo.matches(*item, &minfo);`

Would it be possible to make `ItemTypeInfo::matches()` take the item type instead of storing it in `MaterialInfo`? Possibly even better, `matches()` already takes a `job_item`, which has a `item_type` field per above - if this field is already set to the correct value, that could work (but I'm not sure if it is).

Reply:
I did some investigating on my end, and it looks like ItemTypeInfo::matches is getting a job_item with `item_type = NONE` after creating a metal bar with `createitem BAR IRON`, enabling bars with `b-C-f-G-r`, and planning a new floor. Tables do seem to result in `item_type = TABLE` being set. I probably misunderstood the purpose of job_item - I'm guessing its fields make up more of a template and less of a concrete item.